cors/expr/cors.go

```diff
@@ -37,10 +37,11 @@ type (
 // by origin string) for the given service.
 func Origins(svc string) []*OriginExpr {
 	origins := make(map[string]*OriginExpr)
-	for n, o := range Root.ServiceOrigins {
-		s, ok := o.Parent.(*expr.ServiceExpr)
-		if ok && s.Name == svc {
-			origins[n] = o
+	for s, no := range Root.ServiceOrigins {
```

Comment:
Would it make sense to index the map with the name of the service instead of a reference? Names of services have to be unique and this would alleviate the need for iterating here and instead allow for a direct lookup.

Reply:
Much more sense, changed